server/repository/assignments.go

```diff
@@ -0,0 +1,145 @@
+package repository
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/src-d/code-annotation/server/model"
+)
+
+// Assignments repository
+type Assignments struct {
+	DB *sql.DB
+}
+
+// ErrNoAssignmentsInitialized is the error returned when the Assignments of a
+// User are requested for a given Experiment, but they have not been yet created
+var ErrNoAssignmentsInitialized = fmt.Errorf("No assignments initialized")
+
+// Initialize builds the assignments for the given user and experiment IDs
+func (repo *Assignments) Initialize(userID int, experimentID int) ([]*model.Assignment, error) {
+	tx, err := repo.DB.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	insert, err := tx.Prepare("INSERT INTO assignments VALUES ($1, $2, $3, $4, $5)")
+	if err != nil {
+		return nil, fmt.Errorf("DB error: %v", err)
+	}
+
+	rows, err := repo.DB.Query(
+		"SELECT id FROM file_pairs WHERE experiment_id=$1", experimentID)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting file_pairs from the DB: %v", err)
+	}
+
+	answer := ""
```

Comment:
according to https://github.com/src-d/code-annotation/issues/2 it should be `nil`. Frontend also expects `nil` there.

Reply:
We could use a null value in the DB, but isn't it going to end up as an empty string when we read it back?
https://github.com/src-d/code-annotation/pull/52/files#diff-b01fac8859e3c130c7e1a433072473edR24

Reply:
no if you use `https://golang.org/pkg/database/sql/#NullString`

Reply:
Ok... what about in serializers.go, how does the final json field look like?

Reply:
it should be `*string`

Reply:
done in 07e2f9c